Tests/GitHubReferences.Tests.ps1

```diff
@@ -0,0 +1,171 @@
+# the tests.  We'll restore it at the end.
+$configFile = New-TemporaryFile
+
+try
+{
+    Backup-GitHubConfiguration -Path $configFile
+    Reset-GitHubConfiguration
+    Set-GitHubConfiguration -DisableTelemetry # We don't want UT's to impact telemetry
+    Set-GitHubConfiguration -LogRequestBody # Make it easier to debug UT failures
+
+    if ($accessTokenConfigured)
+    {
+
+        Describe 'Create a new reference(branch) in repository' {
+            $repositoryName = [Guid]::NewGuid()
+            $repo = New-GitHubRepository -RepositoryName $repositoryName -AutoInit
+            $existingref = @(Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference "heads/master")
+            $sha = $existingref.object.sha
+
+            Context 'On creating a valid reference in a new repository from a given SHA' {
+                $refName = "refs/heads/" + [Guid]::NewGuid().ToString()
+                $result = @(New-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $refName -Sha $sha)
+
+                It 'Should successfully create the reference' {
+                    $result.ref | Should Be $refName
+                }
+            }
+
+            Context 'On creating a valid reference in a new repository (specified by Uri) from a given SHA' {
+                $refName = "refs/heads/" + [Guid]::NewGuid().ToString()
+                $result = @(New-GitHubReference -Uri $repo.svn_url -Reference $refName -Sha $sha)
+
+                It 'Should successfully create the reference' {
+                    $result.ref | Should Be $refName
+                }
+            }
+
+            Context 'On creating an existing reference in a new repository from a given SHA' {
+                $refName = "refs/heads/master"
+
+                It 'Should throw an Exception' {
+                    { @(New-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference $refName -Sha $sha) } | Should Throw
+                }
+            }
+
+            Context 'On creating an existing reference in a new repository (specified by Uri) from a given SHA' {
+                $refName = "refs/heads/master"
+
+                It 'Should throw an exception' {
+                    { @(New-GitHubReference -Uri $repo.svn_url -Reference $refName -Sha $sha) }  | Should Throw
+                }
+            }
+
+            $null = Remove-GitHubRepository -OwnerName $ownerName -RepositoryName $repositoryName
+        }
+
+        Describe 'Getting a reference(branch) from repository' {
+            $repositoryName = [Guid]::NewGuid()
+            $repo = New-GitHubRepository -RepositoryName $repositoryName -AutoInit
+
+            Context 'On getting a valid reference from a new repository' {
+                $reference = @(Get-GitHubReference -OwnerName $ownerName -RepositoryName $repositoryName -Reference "heads/master")
```

Comment:
it would probably be worthwhile to have a constant for `head/master` that you use here and elsewhere, and then when testing against it, you could just prepend `refs/` to that same value.